### PR TITLE
feat: require zero address checks

### PR DESCRIPTION
**Pull request by morgante**
This is a helpful check for some classes of problems in Solidity contracts.

### CODE REVIEW

File: .grit/patterns/CheckZero.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+```grit
+language sol
+
+// Marks the body of a function as "checked"
+pattern CheckedBody($address) = bubble($address) contains or {
+    // Directly checks the require
+    `require($address != address(0), $_)`
+    `require($address != address(0))`
+    `require(address(0) != $address, $_)`
+    `require(address(0) != $address)`
```
Comment (ilevyor) [inline .grit/patterns/CheckZero.md:18]: lets get in the habit of adding commas between patterns

File: .grit/patterns/CheckZero.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    // Or look for a function call
+    sol_call_expression(function=`$id`, children=contains $address) where {
+        $program <: contains bubble($id) sol_function_definition(name=$id, children=$child2) where {
+            // Inspect the body of the transitive function recursively
+            $child2 <: CheckedBody($address)
```
Comment (ilevyor) [inline .grit/patterns/CheckZero.md:24]: Should discuss with cos, but iirc current implementation of scopes in Marzano is not conducive to recursive pattern definitions.

File: .grit/patterns/CheckZero.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        // require(from != address(0), "ERC20: transfer from the zero address");
+        // require(to != address(0), "ERC20: transfer to the zero address");
+        uint256 toBalance   = _balances[to];
+        uint256 fromBalance = _balances[from];
+        require(fromBalance >= amount, "ERC20: transfer amount exceeds balance");
+        uint256 fromBalanceUpdated = fromBalance - amount;
+        uint256 toBalanceUpdated   = toBalance   + amount;
+        _balances[from] = fromBalanceUpdated;
+        _balances[to]   = toBalanceUpdated;
+    }
```
Comment (ilevyor) [inline .grit/patterns/CheckZero.md:86]: does this exercise the recursive case?

**Review by ilevyor — APPROVED** [review]
lgtm

File: .grit/patterns/CheckZero.md (hunk 3 above)
Comment (morgante) [reply .grit/patterns/CheckZero.md:86]: Yes.

**Pull request merged**